### PR TITLE
Feature/calculations

### PR DESCRIPTION
**Pull request by ishimwe07**
## Tasks Completed
- [x] add helper functions for formulas 
- [x] add function to calculate the payout amount `payoutCalculation`
- [x] invoke the `payoutCalculation` when Locking a bonus Tranche
- [x] check if the tranche falls in a fiscal year while editing
- [x] assign final amounts when completing a tranche
- [x] Include someone who was excluded before
- [x] Calculate bonus payable for each fiscal year with completed tranches
- [x] All weight for tranches in same fiscal year should not exceed 100% on bonus tranche creation and update

### CODE REVIEW

File: src/handlers/getBonusPayable.action.ts [hunk 1]
```diff
@@ -12,48 +13,68 @@ const logger = cds.log("Participants final payout action.");
...
-            trancheParticipationRecord("*");
-            trancheParticipationRecord.participant("*");
-          }
-        ).where({
-          "bonusTranche.status": TrancheStatusEnum.Completed,
-          "bonusTranche.fiscalYear": fiscalYear,
-        });
+      const allEmployees: Employee[] = await SELECT(
+        Employee,
+        (employeeRecord: QLExtensions<Required<Employee>>) => {
```
Comment (pmy09) [inline src/handlers/getBonusPayable.action.ts:22]: CAP already has a way of handling this scenario:

`await SELECT.from(Employee)
.columns(c => {
c("*"), c.trancheParticipation(participation) => {} ............
})`

File: src/handlers/update-bonusTranche.action.ts [hunk 2]
```diff
@@ -102,12 +142,105 @@ export class UpdateBonusTranche {
...
       throw error;
     }
   }
+
+  private async payoutCalculation(trancheId: string) {
+    cds.spawn({}, async () => {
+      const bonusParticipationRecord = await SELECT.from(
+        TrancheParticipation.name
+      )
+        .columns((bonuses) => [
```
Comment (pmy09) [inline src/handlers/update-bonusTranche.action.ts:169]: You have it implemented here so why not above?

**Review by pmy09 — CHANGES_REQUESTED** [review]
Please check the comment and implement it for consistency

**Review by Isaac-Ampomah-12 — APPROVED** [review]

**Review by pmy09 — APPROVED** [review]

**Review by Isaac-Ampomah-12 — APPROVED** [review]

**Review by Bateyjosue — APPROVED** [review]
## LGTM :tada:

**Pull request merged**